components/insight/SRC/org/openmicroscopy/shoola/env/data/OMEROGateway.java

```diff
@@ -2397,6 +2410,12 @@ private RenderingEnginePrx generateRenderingEngine(
 			service.lookupPixels(pixelsID);
 			needDefault(pixelsID, service);
 			service.load();
+            Set<Long> pixIds = renderingEngines.get(ctx);
```

Comment:
for big images several rendering engines are created for the same set of pixels
Have you considered that case?

Reply:
Shouldn't be a problem, works like before. Only the pixels Id is needed to reload the RenderingEngines, it's only used here: https://github.com/openmicroscopy/openmicroscopy/blob/develop/components/insight/SRC/org/openmicroscopy/shoola/env/data/DataServicesFactory.java#L365 (respectively bit further down at line 388 to reload the RenderingControl for the pixels Id (which should take care of the different RenderingEngines)).